src/frontend/src/views/Data/components/TreeTableNameCell/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from "react";
+import dataTableStyle from "src/common/components/library/data_table/index.module.scss";
+import { RowContent } from "src/common/components/library/data_table/style";
+import { ReactComponent as ExternalLinkIcon } from "src/common/icons/ExternalLink.svg";
+import { ReactComponent as TreeIcon } from "src/common/icons/PhyloTree.svg";
+import { createTreeModalInfo } from "src/common/utils";
+import TreeVizModal from "../TreeVizModal";
+
+interface NameProps {
+  value: string;
+}
+
+const TreeTableNameCell = ({ value }: NameProps): JSX.Element => {
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const treeId = String(value).split(" ")[0];
+
+  return (
+    <RowContent className={dataTableStyle.cell}>
+      <TreeVizModal
```

Comment:
`TreeTableNameCell` component renders `TreeVizModal` and controls whether it's open or not